runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -3224,7 +3230,7 @@ TR::Register *J9::Power::TreeEvaluator::ArrayStoreCHKEvaluator(TR::Node *node, T
    dstReg = cg->evaluate(destinationChild);
    srcReg = cg->evaluate(sourceChild);
 
-   conditions = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(7, 7, cg->trMemory());
+   conditions = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(9, 9, cg->trMemory());
```

Comment:
If I'm understanding correctly, the increase here used by `generateDepLabelInstruction()` is for the possible child node `awrtbari`. I'm just wondering if other nodes that could have `awrtbari` as a child need it too or its a special case for `ArrayStoreCHK`.

Reply:
I believe the only two nodes that can have `awrtbari` as a child are `ArrayStoreCHK` (which, as you pointed out, is taken care of), and `arraycopy`, which uses a different set of instructions that doesn't have the redundant load (defined in `J9::Power::TreeEvaluator::genWrtbarForArrayCopy`).